pkg/kapp/diffgraph/change_graph_test.go

```diff
@@ -590,6 +592,41 @@ roleRef:
 	require.Equal(t, expectedOutput, output)
 }
 
+func TestAppCRSvcAccntRBACDelete(t *testing.T) {
+	configYAML, err := ioutil.ReadFile("assets/appcr-rbac-svcaccnt.yaml")
+	require.NoErrorf(t, err, "Reading appcr-rbac-svcaccnt asset")
+
+	configRs, err := ctlres.NewFileResource(ctlres.NewBytesSource([]byte(configYAML))).Resources()
+	require.NoErrorf(t, err, "Parsing resources")
+
+	rs, conf, err := ctlconf.NewConfFromResourcesWithDefaults(configRs)
+	require.NoErrorf(t, err, "Parsing conf defaults")
+
+	opts := buildGraphOpts{
+		resources:           rs,
+		op:                  ctldgraph.ActualChangeOpDelete,
+		changeGroupBindings: conf.ChangeGroupBindings(),
+		changeRuleBindings:  conf.ChangeRuleBindings(),
+	}
+
+	t1 := time.Now()
+
+	graph, err := buildChangeGraphWithOpts(opts, t)
+	require.NoErrorf(t, err, "Expected graph to build")
+
+	require.Less(t, time.Now().Sub(t1), time.Duration(1*time.Second), "Graph build took too long")
+
+	output := strings.TrimSpace(graph.PrintLinearizedStr())
```

Comment:
Should we use `graph.PrintStr()` like other tests?

Reply:
Agree. 2nd one looks more readable.

Reply:
With the first output we get to know that on which change(s) a particular change depends, but the second output only shows the various groups that are formed in the change graph which sometimes is sufficient and sometimes not.
In this particular example it is evident that the all the resources in the second group depend on the resource present in the first group (since there is only one resource), so it is fine if we want to go ahead with it.